query_optimizer/QueryHandle.hpp

```diff
@@ -178,6 +178,14 @@ class QueryHandle {
     query_result_relation_ = relation;
   }
 
+  /**
+   * @brief Return all the base relations referenced in this query.
+   **/
+  std::vector<relation_id> getAllReferencedBaseRelations() {
```

Comment:
This should mark as a `const` method, and I think we need to add a data member called `referenced_base_relations_`.